simple_graph.py

```diff
@@ -89,8 +89,8 @@ def breadth_first_traversal(self, node):
          nodes in the order visited."""
         visited = []
         q = [node]
-        while q:
-            vertex = q.pop()
+        while len(q) > 0:
+            vertex = q.pop(0)
```

Comment:
Use a real queue, either the one you wrote, or the one in the stdlib `collections` module:  `collections.deque`